test/langtools/tools/javac/classfiles/attributes/innerclasses/InnerClassesInLocalClassTest.java

```diff
@@ -52,7 +52,7 @@ public InnerClassesInLocalClassTest() {
         for (Modifier outerModifier : LOCAL_CLASS_MODIFIERS) {
             StringBuilder sb = new StringBuilder();
             sb.append(outerModifier.getString()).append(' ');
-            sb.append("class Local {");
+            sb.append("class Local { int f; ");
             Map<String, Set<String>> class2Flags = new HashMap<>();
```

Comment:
Hack to suppress ACC_PERMITS_VALUE bit from being set thereby causing the test to "fail"

Reply:
I was playing around to understand this generated test:

```
public class OuterClass {
 static void method() {abstract class Local { class A0{}
abstract class A1{}
final class A2{}
};}
}
```

Why do we get the following for an inner class that is abstract,

```
[ASSERT] : inner_class_access_flags 1Local
Expected: [ACC_ABSTRACT],
     Got: [ACC_ABSTRACT, 0x40]
```

while `abstract class A8` in `PermitsValueTest.java` does not emit `ACC_PERMITS_VALUE`?

nitpick (and possibly not addressed yet): `Modifiers:getString()` doesn't return the name for `ACC_PERMITS_VALUE` yet. Do we need to address this in this PR? possibly not

Reply:
> I was playing around to understand this generated test:
> 
> ```
> public class OuterClass {
>  static void method() {abstract class Local { class A0{}
> abstract class A1{}
> final class A2{}
> };}
> }
> ```
> 
> Why do we get the following for an inner class that is abstract,
> 
> ```
> [ASSERT] : inner_class_access_flags 1Local
> Expected: [ACC_ABSTRACT],
>      Got: [ACC_ABSTRACT, 0x40]
> ```
> 
> while `abstract class A8` in `PermitsValueTest.java` does not emit `ACC_PERMITS_VALUE`?

Class A8 in PermitsValue is an inner class that has an enclosing instance. Enclosing instance means the class will have an instance field to store the outer this and also its constructor will receive an argument - ie it cannot be a no-arg constructor only. (You may want to see the class files generated to understand the synthetic fields and constructor args that represent the outer this for an inner class with an enclosing instance) 

> 
> nitpick (and possibly not addressed yet): `Modifiers:getString()` doesn't return the name for `ACC_PERMITS_VALUE` yet. Do we need to address this in this PR? possibly not

Oops - Yes I will raise a follow up defect.

Reply:
Thank you! It's clear now!